frame/system/src/lib.rs

```diff
@@ -171,12 +171,12 @@ pub trait Trait: 'static + Eq + Clone {
 	/// The block number type used by the runtime.
 	type BlockNumber:
 		Parameter + Member + MaybeSerializeDeserialize + Debug + MaybeDisplay + SimpleArithmetic
-		+ Default + Bounded + Copy + sp_std::hash::Hash + sp_std::str::FromStr;
+		+ Default + Bounded + Copy + sp_std::hash::Hash + sp_std::str::FromStr + MaybeMallocSizeOf;
 
 	/// The output of the `Hashing` function.
 	type Hash:
 		Parameter + Member + MaybeSerializeDeserialize + Debug + MaybeDisplay + SimpleBitOps + Ord
-		+ Default + Copy + CheckEqual + sp_std::hash::Hash + AsRef<[u8]> + AsMut<[u8]>;
+		+ Default + Copy + CheckEqual + sp_std::hash::Hash + AsRef<[u8]> + AsMut<[u8]> + MaybeMallocSizeOf;
```

Comment:
Hmm, that's a pretty strong requirement to put into the runtime imho, requires bringing in a separate (parity-specific) lib to implement properly. Can't we add `where` clauses instead?

Reply:
Unless some exotic Hash type is used, it is implemented by default within `parity-util-mem` (all H*, integers, fixed arrays, `Vec<u8>` etc. are covered)

Reply:
And we have `MaybeSerializeDeserialize` there,, so using exotic Hash will require some work anyway.

Reply:
@bkchr what's your view on this?
@tomusdrw are you ok with an argument?

Reply:
For me this is okay.